ambry-router/src/main/java/com.github.ambry.router/GetBlobOperation.java

```diff
@@ -560,9 +592,9 @@ private void fetchRequests(RequestRegistrationCallback<GetOperation> requestRegi
      * Check if the operation on the chunk is eligible for completion, if so complete it.
      */
     void checkAndMaybeComplete() {
-      if (chunkOperationTracker.isDone()) {
+      if (!chunkCompleted && chunkOperationTracker.isDone()) {
```

Comment:
Added a new flag `retainOperationOnSuccess` that will be set to `true` when a set exception should be reported even if the operation tracker reports a success.  `chunkCompleted` now only signifies if the operation has been completed or not.